File: Sorting-Algorithms/Bubble-Sort/bubbleSort.py
"""
Bubble sort is an algorithm which steps through the array and if an elements is out of order, swap elements

Time Complexity is O(n^2)

"""

def bubbleSort(array):
   is_sorted = False
   last_unsorted = len(array)-1

   while not is_sorted:
      is_sorted = True
      for i in range(last_unsorted):
         if array[i] > array[i+1]:
            array[i], array[i+1] = array[i+1], array[i]
            is_sorted = False
      last_unsorted -= 1

File: Algorithms/Sorting-Algorithms/Bubble-Sort/bubbleSort.cpp
#include<iostream>
#include<vector>

void bubbleSort(std::vector<int>& array){
   bool isSorted = false;
   int lastUnsorted = array.size()-1;

   while(isSorted == false){
      isSorted = true;
      for(int i = 0; i < lastUnsorted; i++){
         if(array[i] > array[i+1]){
            int temp = array[i];
            array[i] = array[i+1];
            array[i+1] = temp;
            isSorted = false;
         }
      }
      lastUnsorted--;
   }
}